iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -387,22 +415,59 @@ static void legalizeDispatchWorkgroupOperands(
       // TODO(nicolasvasilache): do something useful.
       continue;
     }
-    operand.replaceAllUsesExcept(
-        repl, SmallPtrSet<Operation *, 8>(uses.begin(), uses.end()));
+    map.map(operand, repl);
   }
 
-  // Reinsert and replace old BB args.
-  for (BlockArgument ba : block.getArguments().take_front(numOldBBArgs)) {
-    block.addArgument(ba.getType());
-    ba.replaceAllUsesWith(block.getArguments().back());
+  // The only existing argument are for the outputs. Just need to add a new
+  // argument for the outpts and remap the value to use the new argument.
+  for (auto ba : block.getArguments().take_front(numOldBBArgs)) {
+    assert(ba.getType().isa<IREE::Flow::DispatchOutputType>());
+    map.map(ba, block.addArgument(ba.getType()));
+  }
+
+  auto getUsesOfValueOutsideOfDispatchOp =
+      [&](Value v) -> SmallPtrSet<Operation *, 4> {
```

Comment:
Nit: I think there is no need to explicitly list `-> SmallPtrSet<Operation *, 4>`? The logic of this lambda is fairly simple so missing that sounds okay to me.